diff --git a/includes/entity.inc b/includes/entity.inc
index 25f7584..384e51e 100644
--- a/includes/entity.inc
+++ b/includes/entity.inc
@@ -361,7 +361,7 @@ class DrupalDefaultEntityController implements DrupalEntityControllerInterface {
     if ($conditions) {
       foreach ($entities as $entity) {
         $entity_values = (array) $entity;
-        if (array_diff_assoc($conditions, $entity_values)) {
+        if (drupal_array_diff_assoc_recursive($conditions, $entity_values)) {
           unset($entities[$entity->{$this->idKey}]);
         }
       }
